src/clib/cool1d_multi_g.F

```diff
@@ -485,13 +773,44 @@ subroutine cool1d_multi_g(
       enddo
 
       endif
-     
+
 !     --- H2 cooling ---
 
       if (ispecies .gt. 1) then
+!        Chiaki & Wise (2019) H2 cooling rate
+         if (ih2cr .eq. 3) then
+
+      do i = is+1, ie+1
+      if ( itmask(i) ) then
+
+         lognhat = logH2I(i) - logdvdr(i)
+
+         call interpolate_3D_g(
+     &     lognhat, logT(i), logH(i), int8(LH2_N), 
+     &     LH2_D, LH2_dD, LH2_T, LH2_dT, LH2_H, LH2_dH,
+     &     int8(LH2_Size), LH2_L, log_Linv)
+         L = 1.d1**(-log_Linv)
+
+         if (icmbTfloor .eq. 1) then
+         call interpolate_3D_g(
+     &     lognhat, logTcmb(i), logH(i), int8(LH2_N), 
+     &     LH2_D, LH2_dD, LH2_T, LH2_dT, LH2_H, LH2_dH,
+     &     int8(LH2_Size), LH2_L, log_Ginv)
+         G = 1.d1**(-log_Ginv)
+         else
+         G = tiny8
+         endif
+
+         LH2(i) = - (L - G) / dom * H2I(i,j,k)/2.d0
+         if (LH2(i) .ne. LH2(i)) LH2(i) = 0.d0
+         edot(i) = edot(i) + ih2cox*LH2(i)
```

Comment:
The `ih2cox` should be applied to LH2 instead of edot.
```suggestion
         LH2(i) = - ih2cox * (L - G) / dom * H2I(i,j,k)/2.d0
         if (LH2(i) .ne. LH2(i)) LH2(i) = 0.d0
         edot(i) = edot(i) + LH2(i)
```